common/src/json_path_writer.rs

```diff
@@ -40,7 +54,9 @@ impl JsonPathWriter {
         let len_path = self.path.len();
         self.indices.push(len_path);
         if !self.path.is_empty() {
-            self.path.push_str(JSON_PATH_SEGMENT_SEP_STR);
+            unsafe {
+                self.path.as_mut_vec().push(JSON_PATH_SEGMENT_SEP);
```

Comment:
is the unsafe changing anything?

Reply:
https://godbolt.org/z/s1bf4s9zW

append_char does the same as byte.

Reply:
good point, removed the unsafe